client/web/src/LegacyLayout.tsx

```diff
@@ -154,10 +159,30 @@ export const LegacyLayout: FC<LegacyLayoutProps> = props => {
     // setup wizard state, since we don't have a good solution for this at the
     // moment, we use mutable window.context object here.
     // TODO remove window.context and use injected context store/props
-    if (needsRepositoryConfiguration && !wasSetupWizardSkipped && props.authenticatedUser?.siteAdmin) {
+    if (
+        !props.isSourcegraphApp &&
+        needsRepositoryConfiguration &&
+        !wasSetupWizardSkipped &&
+        props.authenticatedUser?.siteAdmin
+    ) {
         return <Navigate to={PageRoutes.SetupWizard} replace={true} />
     }
 
+    // Redirect to the app setup pages if it's App, we haven't seen setup before,
+    // and it's not already a setup page
+    if (props.isSourcegraphApp && !wasAppSetupFinished && !isAppSetupPage) {
+        return <Navigate to={EnterprisePageRoutes.AppSetup} replace={true} />
+    }
+
+    // Some route by their design requires rendering on the blank page
+    // without some general and standard UI that Layout component renders
+    // by default. If route has handle: { fullPage: true } we render just
+    // routes and its container block without rendering global nav, notepad
+    // and other standard UI elements.
```

Comment:
minor wording suggestions

```suggestion
    // Some routes by their design require rendering on a blank page
    // without the UI chrome that Layout component renders by default.
    // If route has handle: { fullPage: true } we render just the route content
    // and its container block without rendering global nav, notepad
    // and other standard UI chrome elements.
```